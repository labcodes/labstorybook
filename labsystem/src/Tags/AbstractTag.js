import React from "react";
import PropTypes from "prop-types";
import { isUndefined } from "lodash";

export default class AbstractTag extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    isOutline: PropTypes.bool,
    isOn: PropTypes.bool,
    thumbSrc: PropTypes.string,
    icon: PropTypes.string,
    skin: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func,
    renderPrefix: PropTypes.object,
    renderSuffix: PropTypes.object,
    className: PropTypes.string,
  };

  static defaultProps = {
    thumbSrc: "",
    icon: "",
    isOutline: false,
    skin: "pale",
    color: "",
    disabled: false,
    isOn: false,
    onClick: undefined,
    renderPrefix: undefined,
    renderSuffix: undefined,
    className: "",
  };

  handleOnClick = (e) => {
    const { onClick } = this.props;
    if (!isUndefined(onClick)) {
      onClick(e);
    }
  };

  render() {
    const {
      text,
      disabled,
      icon,
      thumbSrc,
      isOutline,
      skin,
      color,
      isOn,
      renderPrefix,
      renderSuffix,
      className,
    } = this.props;

    return (
      <span
        className={
          `lab-tag ${className}` +
          `${disabled ? ` lab-tag--disabled` : ""}` +
          `${isOutline ? ` lab-tag--outline` : ""}` +
          `${color ? ` lab-tag--${color}-${skin}` : ` lab-tag--${skin}`}` +
          `${isOn ? ` lab-tag--selected` : ""}` +
          `${icon ? ` lab-tag--has-left-icon` : ""}` +
          `${thumbSrc ? ` lab-tag--has-thumb` : ""}`
        }
        onClick={this.handleOnClick}
        onKeyDown={this.handleOnClick}
        role="presentation"
      >
        {renderPrefix}
        {text}
        {renderSuffix}
      </span>
    );
  }
}
