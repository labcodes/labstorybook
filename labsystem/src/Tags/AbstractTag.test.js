import React from "react";
import renderer from "react-test-renderer";

import AbstractTag from "./AbstractTag";

describe("AbstractTag", () => {
  it("renders with base props for 'simple' type", async () => {
    const renderedComponent = renderer
      .create(<AbstractTag text="Test SimpleTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with base props for 'removable' type", async () => {
    const renderedComponent = renderer
      .create(<AbstractTag removable text="Test RemovableTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with base props for 'togglable' type", async () => {
    const renderedComponent = renderer
      .create(<AbstractTag isTogglable text="Test TogglableTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with base props for 'dropdown' type", async () => {
    const renderedComponent = renderer
      .create(<AbstractTag isDropdown text="Test DropdownTag" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
