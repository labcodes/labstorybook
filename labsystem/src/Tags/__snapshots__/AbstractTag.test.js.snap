// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractTag renders with base props for 'dropdown' type 1`] = `
<span
  className="lab-tag  lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test DropdownTag
</span>
`;

exports[`AbstractTag renders with base props for 'removable' type 1`] = `
<span
  className="lab-tag  lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test RemovableTag
</span>
`;

exports[`AbstractTag renders with base props for 'simple' type 1`] = `
<span
  className="lab-tag  lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test SimpleTag
</span>
`;

exports[`AbstractTag renders with base props for 'togglable' type 1`] = `
<span
  className="lab-tag  lab-tag--pale"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  Test TogglableTag
</span>
`;
